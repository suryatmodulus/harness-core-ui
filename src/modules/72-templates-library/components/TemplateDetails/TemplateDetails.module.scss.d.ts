/* eslint-disable */
// this is an auto-generated file
declare const styles: {
  readonly container: string
  readonly shrink: string
  readonly openInStudio: string
  readonly topContainer: string
  readonly tabsContainer: string
  readonly innerTabsContainer: string
  readonly dropdown: string
}
export default styles
