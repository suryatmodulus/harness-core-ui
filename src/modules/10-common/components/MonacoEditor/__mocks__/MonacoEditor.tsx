/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import type { MonacoEditorProps } from 'react-monaco-editor'

export default function MonacoEditor(
  props: MonacoEditorProps & { name: string; value?: string; 'data-testid'?: string }
) {
  return (
    <textarea
      name={props.name}
      value={props.value}
      data-testid={props['data-testid']}
      onChange={e => props.onChange?.(e.target.value, e as any)}
    />
  )
}
