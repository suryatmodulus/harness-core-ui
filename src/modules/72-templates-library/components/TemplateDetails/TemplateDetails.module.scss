.container {
  position: relative;
  overflow-wrap: anywhere;
  overflow: auto;
  .shrink {
    flex-shrink: 1 !important;
  }
  .openInStudio {
    flex-shrink: 0;
  }
  .topContainer {
    p {
      --font-size: var(--form-input-font-size) !important;
    }
    :global {
      .DropDown--main {
        display: flex;
      }
      .DropDown--dropdownButton {
        background-color: var(--white) !important;
      }
    }
  }
  .tabsContainer {
    flex-grow: 1;
    overflow: hidden;

    :global {
      .bp3-tabs {
        height: 100%;
      }
      .bp3-tab-panel {
        margin-top: 0px !important;
        height: calc(100% - 43px);
        overflow: auto;
      }
      .bp3-tab-list {
        background-color: var(--white);
        padding: 0 var(--spacing-xlarge);
        box-shadow: var(--elevation-2);
        z-index: 1;
      }
      .bp3-tab {
        font-size: 12px !important;
        margin-right: var(--spacing-5) !important;
      }
    }
  }
  .innerTabsContainer {
    :global {
      .bp3-tab-panel {
        overflow: unset;
        height: auto;
      }
      .bp3-tab-list {
        border-top: 1px solid var(--grey-200);
        position: sticky;
        top: 0;
      }
    }
  }
  .dropdown {
    :global(.bp3-menu) {
      width: 300px !important;
    }
  }
}
