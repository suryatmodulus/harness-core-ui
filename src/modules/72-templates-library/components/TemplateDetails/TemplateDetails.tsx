import React, { useContext } from 'react'
import {
  ButtonSize,
  ButtonVariation,
  Color,
  Container,
  DropDown,
  Icon,
  Layout,
  PageError,
  SelectOption,
  Tab,
  Tabs,
  Text
} from '@wings-software/uicore'
import { useHistory, useParams } from 'react-router-dom'
import { defaultTo, isEmpty } from 'lodash-es'
import { useStrings } from 'framework/strings'
import routes from '@common/RouteDefinitions'
import { TemplateTags } from '@templates-library/components/TemplateTags/TemplateTags'
import { PageSpinner } from '@common/components'
import {
  getIconForTemplate,
  getTypeForTemplate,
  TemplateListType
} from '@templates-library/pages/TemplatesPage/TemplatesPageUtils'
import { useMutateAsGet } from '@common/hooks'
import { TemplateSummaryResponse, useGetTemplateList } from 'services/template-ng'
import RbacButton from '@rbac/components/Button/Button'
import { ResourceType } from '@rbac/interfaces/ResourceType'
import { PermissionIdentifier } from '@rbac/interfaces/PermissionIdentifier'
import { useAppStore } from 'framework/AppStore/AppStoreContext'
import GitPopover from '@pipeline/components/GitPopover/GitPopover'
import type { ModulePathParams, ProjectPathProps } from '@common/interfaces/RouteInterfaces'
import { TemplateContext } from '../TemplateStudio/TemplateContext/TemplateContext'
import { TemplateInputs } from '../TemplateInputs/TemplateInputs'
import { TemplateYaml } from '../TemplateYaml/TemplateYaml'
import { TemplateActivityLog } from '../TemplateActivityLog/TemplateActivityLog'
import css from './TemplateDetails.module.scss'

export interface TemplateDetailsProps {
  template: TemplateSummaryResponse
  allowStableSelection?: boolean
  setTemplate?: (template: TemplateSummaryResponse) => void
}

export enum TemplateTabs {
  INPUTS = 'INPUTS',
  YAML = 'YAML',
  REFERENCEDBY = 'REFERENCEDBY'
}

export enum ParentTemplateTabs {
  BASIC = 'BASIC',
  ACTVITYLOG = 'ACTVITYLOG'
}

const DefaultStableVersionValue = '-1'

export const TemplateDetails: React.FC<TemplateDetailsProps> = props => {
  const { template, allowStableSelection = false, setTemplate } = props
  const { getString } = useStrings()
  const history = useHistory()
  const [versionOptions, setVersionOptions] = React.useState<SelectOption[]>([])
  const { isReadonly } = useContext(TemplateContext)
  const { isGitSyncEnabled } = useAppStore()
  const [templates, setTemplates] = React.useState<TemplateSummaryResponse[]>([])
  const [selectedTemplate, setSelectedTemplate] = React.useState<TemplateSummaryResponse>()
  const [selectedParentTab, setSelectedParentTab] = React.useState<ParentTemplateTabs>(ParentTemplateTabs.BASIC)
  const [selectedTab, setSelectedTab] = React.useState<TemplateTabs>(TemplateTabs.YAML)
  const { module } = useParams<ProjectPathProps & ModulePathParams>()

  const {
    data: templateData,
    refetch: reloadTemplates,
    loading,
    error: templatesError
  } = useMutateAsGet(useGetTemplateList, {
    body: {
      filterType: 'Template',
      templateIdentifiers: [template.identifier]
    },
    queryParams: {
      accountIdentifier: template.accountId,
      orgIdentifier: template.orgIdentifier,
      projectIdentifier: template.projectIdentifier,
      templateListType: TemplateListType.All,
      module,
      repoIdentifier: template.gitDetails?.repoIdentifier,
      branch: template.gitDetails?.branch
    },
    queryParamStringifyOptions: { arrayFormat: 'comma' }
  })

  const onChange = React.useCallback(
    (option: SelectOption): void => {
      const version = defaultTo(option.value?.toString(), '')
      if (version === DefaultStableVersionValue) {
        setSelectedTemplate(templates.find(item => !item.versionLabel))
      } else {
        setSelectedTemplate(templates.find(item => item.versionLabel === version))
      }
    },
    [templates]
  )

  React.useEffect(() => {
    if (selectedTemplate) {
      setTemplate?.(selectedTemplate)
    }
  }, [selectedTemplate])

  React.useEffect(() => {
    const newVersionOptions: SelectOption[] = templates.map(item => {
      return {
        label: isEmpty(item.versionLabel)
          ? getString('templatesLibrary.alwaysUseStableVersion')
          : item.stableTemplate
          ? getString('templatesLibrary.stableVersion', { entity: item.versionLabel })
          : item.versionLabel,
        value: defaultTo(item.versionLabel, DefaultStableVersionValue)
      } as SelectOption
    })
    setVersionOptions(newVersionOptions)
  }, [templates])

  React.useEffect(() => {
    const allVersions = [...(templateData?.data?.content || [])]
    if (allowStableSelection) {
      const stableVersion = { ...allVersions.find(item => item.stableTemplate) }
      if (stableVersion) {
        delete stableVersion.versionLabel
        allVersions.unshift(stableVersion)
      }
    }
    setTemplates(allVersions)
    setSelectedTemplate(allVersions.find(item => item.versionLabel === template.versionLabel))
  }, [templateData?.data?.content])

  const goToTemplateStudio = () => {
    if (selectedTemplate) {
      history.push(
        routes.toTemplateStudio({
          projectIdentifier: selectedTemplate.projectIdentifier,
          orgIdentifier: selectedTemplate.orgIdentifier,
          accountId: defaultTo(selectedTemplate.accountId, ''),
          module,
          templateType: selectedTemplate.templateEntityType,
          templateIdentifier: selectedTemplate.identifier,
          versionLabel: selectedTemplate.versionLabel,
          repoIdentifier: selectedTemplate.gitDetails?.repoIdentifier,
          branch: selectedTemplate.gitDetails?.branch
        })
      )
    }
  }

  const handleTabChange = React.useCallback((tab: TemplateTabs) => {
    setSelectedTab(tab)
  }, [])

  const handleParentTabChange = React.useCallback(
    (tab: ParentTemplateTabs) => {
      setSelectedParentTab(tab)
    },
    [setSelectedParentTab]
  )

  return (
    <Container height={'100%'} className={css.container} data-template-id={template.identifier}>
      <Layout.Vertical flex={{ align: 'center-center' }} height={'100%'}>
        {loading && <PageSpinner />}
        {!loading && templatesError && (
          <PageError
            message={defaultTo((templatesError.data as Error)?.message, templatesError.message)}
            onClick={reloadTemplates}
          />
        )}
        {!templatesError && selectedTemplate && (
          <Container height={'100%'} width={'100%'}>
            <Layout.Vertical height={'100%'}>
              <Layout.Horizontal
                flex={{ alignItems: 'center' }}
                spacing={'huge'}
                padding={{ top: 'large', left: 'xxlarge', bottom: 'large', right: 'xxlarge' }}
                border={{ bottom: true }}
              >
                <Layout.Horizontal className={css.shrink} spacing={'small'}>
                  <Text lineClamp={2} font={{ size: 'medium', weight: 'bold' }} color={Color.GREY_800}>
                    {selectedTemplate.name}
                  </Text>
                  {isGitSyncEnabled && (
                    <GitPopover
                      data={defaultTo(selectedTemplate.gitDetails, {})}
                      iconProps={{ margin: { left: 'small', top: 'xsmall' } }}
                    />
                  )}
                </Layout.Horizontal>
                <RbacButton
                  text={getString('templatesLibrary.openInTemplateStudio')}
                  variation={ButtonVariation.SECONDARY}
                  size={ButtonSize.SMALL}
                  className={css.openInStudio}
                  onClick={goToTemplateStudio}
                  permission={{
                    permission: PermissionIdentifier.VIEW_TEMPLATE,
                    resource: {
                      resourceType: ResourceType.TEMPLATE
                    }
                  }}
                />
              </Layout.Horizontal>
              <Container background={Color.FORM_BG} className={css.tabsContainer}>
                <Tabs id="template-details-parent" selectedTabId={selectedParentTab} onChange={handleParentTabChange}>
                  <Tab
                    id={ParentTemplateTabs.BASIC}
                    title={getString('details')}
                    panel={
                      <Layout.Vertical>
                        <Container>
                          <Layout.Vertical
                            className={css.topContainer}
                            spacing={'large'}
                            padding={{ top: 'xlarge', right: 'xxlarge', bottom: 'xlarge', left: 'xxlarge' }}
                          >
                            <Container>
                              <Layout.Vertical spacing={'small'}>
                                <Text font={{ weight: 'semi-bold' }} color={Color.BLACK}>
                                  {getString('typeLabel')}
                                </Text>
                                <Container>
                                  <Layout.Horizontal
                                    spacing={'small'}
                                    flex={{ alignItems: 'center', justifyContent: 'flex-start' }}
                                  >
                                    <Icon name={getIconForTemplate(selectedTemplate, getString)} size={20} />
                                    <Text color={Color.GREY_900}>
                                      {getTypeForTemplate(selectedTemplate, getString)}
                                    </Text>
                                  </Layout.Horizontal>
                                </Container>
                              </Layout.Vertical>
                            </Container>
                            <Container>
                              <Layout.Vertical spacing={'small'}>
                                <Text font={{ weight: 'semi-bold' }} color={Color.BLACK}>
                                  {getString('description')}
                                </Text>
                                <Text color={Color.GREY_900}>{selectedTemplate.description || '-'}</Text>
                              </Layout.Vertical>
                            </Container>
                            <Container>
                              <Layout.Vertical spacing={'small'}>
                                <Text font={{ weight: 'semi-bold' }} color={Color.BLACK}>
                                  {getString('tagsLabel')}
                                </Text>
                                {selectedTemplate.tags && !isEmpty(selectedTemplate.tags) ? (
                                  <Container>
                                    <TemplateTags tags={selectedTemplate.tags} />
                                  </Container>
                                ) : (
                                  <Text color={Color.GREY_900}>-</Text>
                                )}
                              </Layout.Vertical>
                            </Container>
                            <Container>
                              <Layout.Vertical spacing={'small'}>
                                <Text font={{ weight: 'semi-bold' }} color={Color.BLACK}>
                                  {getString('templatesLibrary.createNewModal.versionLabel')}
                                </Text>
                                <DropDown
                                  filterable={false}
                                  items={versionOptions}
                                  value={defaultTo(selectedTemplate.versionLabel, DefaultStableVersionValue)}
                                  onChange={onChange}
                                  disabled={isReadonly}
                                  width={300}
                                  popoverClassName={css.dropdown}
                                />
                              </Layout.Vertical>
                            </Container>
                          </Layout.Vertical>
                        </Container>
                        <Container className={css.innerTabsContainer}>
                          <Tabs id="template-details" selectedTabId={selectedTab} onChange={handleTabChange}>
                            <Tab
                              id={TemplateTabs.INPUTS}
                              title={getString('templatesLibrary.templateInputs')}
                              panel={<TemplateInputs template={selectedTemplate} />}
                            />
                            <Tab
                              id={TemplateTabs.YAML}
                              title={getString('yaml')}
                              panel={<TemplateYaml templateYaml={selectedTemplate.yaml} />}
                            />
                            <Tab
                              id={TemplateTabs.REFERENCEDBY}
                              disabled={true}
                              title={getString('templatesLibrary.referencedBy')}
                            />
                          </Tabs>
                        </Container>
                      </Layout.Vertical>
                    }
                  />
                  <Tab
                    id={ParentTemplateTabs.ACTVITYLOG}
                    title={getString('activityLog')}
                    panel={<TemplateActivityLog template={selectedTemplate} />}
                  />
                </Tabs>
              </Container>
            </Layout.Vertical>
          </Container>
        )}
      </Layout.Vertical>
    </Container>
  )
}
