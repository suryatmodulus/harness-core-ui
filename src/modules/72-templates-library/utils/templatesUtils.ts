import type { UseStringsReturn } from 'framework/strings'

export enum TemplateType {
  Step = 'Step',
  Stage = 'Stage',
  Pipeline = 'Pipeline',
  Service = 'Service',
  Infrastructure = 'Infrastructure',
  StepGroup = 'StepGroup',
  Execution = 'Execution'
}

export const AllTemplatesTypes = 'All'

export interface TemplateTypeOption {
  label: string
  value: string
  disabled?: boolean
}

export const getAllowedTemplateTypes = (getString: UseStringsReturn['getString']): TemplateTypeOption[] => [
  {
    label: getString('step'),
    value: TemplateType.Step
  },
  {
    label: getString('templatesLibrary.stageTemplate'),
    value: TemplateType.Stage
  },
  {
    label: getString('common.pipeline'),
    value: TemplateType.Pipeline,
    disabled: true
  },
  {
    label: getString('service'),
    value: TemplateType.Service,
    disabled: true
  },
  {
    label: getString('infrastructureText'),
    value: TemplateType.Infrastructure,
    disabled: true
  },
  {
    label: getString('stepGroup'),
    value: TemplateType.StepGroup,
    disabled: true
  },
  {
    label: getString('executionText'),
    value: TemplateType.Execution,
    disabled: true
  }
]
