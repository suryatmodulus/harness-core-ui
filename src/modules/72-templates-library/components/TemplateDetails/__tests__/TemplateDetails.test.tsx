/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { act, fireEvent, render } from '@testing-library/react'
import { useLocation } from 'react-router-dom'
import { TestWrapper } from '@common/utils/testUtils'
import { useMutateAsGet } from '@common/hooks'
import MonacoEditor from '@common/components/MonacoEditor/__mocks__/MonacoEditor'
import { mockTemplates, mockTemplatesSuccessResponse } from '@templates-library/TemplatesTestHelper'
import { TemplateDetails } from '../TemplateDetails'

jest.mock('@common/components/YAMLBuilder/YamlBuilder')
jest.mock('@wings-software/monaco-yaml/lib/esm/languageservice/yamlLanguageService', () => ({
  getLanguageService: jest.fn()
}))

jest.mock('@common/hooks', () => ({
  ...(jest.requireActual('@common/hooks') as any),
  useMutateAsGet: jest.fn()
}))

jest.mock('react-monaco-editor', () => ({
  MonacoDiffEditor: MonacoEditor
}))

jest.mock('@common/components/MonacoEditor/MonacoEditor', () => MonacoEditor)

function ComponentWrapper(): React.ReactElement {
  const location = useLocation()
  return (
    <React.Fragment>
      <TemplateDetails template={mockTemplates?.data?.content?.[0] || {}} />
      <div data-testid="location">{`${location.pathname}${location.search}`}</div>
    </React.Fragment>
  )
}

describe('<TemplateDetails /> tests', () => {
  beforeEach(() => {
    // eslint-disable-next-line
    // @ts-ignore
    useMutateAsGet.mockReturnValue(mockTemplatesSuccessResponse)
  })
  test('snapshot test', async () => {
    const { container } = render(
      <TestWrapper>
        <ComponentWrapper />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })

  test('should open template studio on clicking open in template studio', async () => {
    const { getByRole, getByTestId } = render(
      <TestWrapper>
        <ComponentWrapper />
      </TestWrapper>
    )

    await act(async () => {
      fireEvent.click(getByRole('button', { name: 'templatesLibrary.openInTemplateStudio' }))
    })
    expect(getByTestId('location')).toMatchInlineSnapshot(`
      <div
        data-testid="location"
      >
        /account/kmpySmUISimoRrJL6NL73w/home/orgs/default/projects/Templateproject/setup/resources/template-studio/Step/template/manjutesttemplate/?versionLabel=v4
      </div>
    `)
  })
})
