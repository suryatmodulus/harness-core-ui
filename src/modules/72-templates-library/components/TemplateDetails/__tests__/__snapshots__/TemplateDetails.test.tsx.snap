// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TemplateDetails /> tests snapshot test 1`] = `
<div>
  <div
    class="StyledProps--font StyledProps--main container"
    data-template-id="manjutesttemplate"
    style="height: 100%;"
  >
    <div
      class="Layout--vertical StyledProps--main StyledProps--flex StyledProps--flex-align-center-center"
      style="height: 100%;"
    >
      <div
        class="StyledProps--font StyledProps--main"
        style="width: 100%; height: 100%;"
      >
        <div
          class="Layout--vertical StyledProps--main"
          style="height: 100%;"
        >
          <div
            class="Layout--horizontal Layout--layout-spacing-huge StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center StyledProps--padding-top-large StyledProps--padding-left-xxlarge StyledProps--padding-bottom-large StyledProps--padding-right-xxlarge StyledProps--border StyledProps--border-bottom"
          >
            <div
              class="Layout--horizontal Layout--layout-spacing-small StyledProps--main shrink"
            >
              <p
                class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--font-size-medium StyledProps--font-weight-bold StyledProps--color StyledProps--color-grey800"
                style="--text-line-clamp: 2;"
              >
                manju-test-template-qq-12344
              </p>
            </div>
            <button
              class="bp3-button Button--button StyledProps--font StyledProps--main openInStudio Button--with-current-color Button--variation Button--variation-secondary Button--size Button--small"
              type="button"
            >
              <span
                class="bp3-button-text"
              >
                templatesLibrary.openInTemplateStudio
              </span>
            </button>
          </div>
          <div
            class="StyledProps--font StyledProps--main tabsContainer StyledProps--background-form"
          >
            <div
              class="bp3-tabs Tabs--main Tabs--horizontal"
            >
              <div
                class="bp3-tab-list"
                role="tablist"
              >
                <div
                  class="bp3-tab-indicator-wrapper"
                  style="height: 0px; transform: translateX(0px) translateY(0px); width: 0px;"
                >
                  <div
                    class="bp3-tab-indicator"
                  />
                </div>
                <div
                  aria-controls="bp3-tab-panel_template-details-parent_BASIC"
                  aria-disabled="false"
                  aria-expanded="true"
                  aria-selected="true"
                  class="bp3-tab"
                  data-tab-id="BASIC"
                  id="bp3-tab-title_template-details-parent_BASIC"
                  role="tab"
                  tabindex="0"
                >
                  details
                </div>
                <div
                  aria-controls="bp3-tab-panel_template-details-parent_ACTVITYLOG"
                  aria-disabled="false"
                  aria-expanded="false"
                  aria-selected="false"
                  class="bp3-tab"
                  data-tab-id="ACTVITYLOG"
                  id="bp3-tab-title_template-details-parent_ACTVITYLOG"
                  role="tab"
                  tabindex="0"
                >
                  activityLog
                </div>
              </div>
              <div
                aria-hidden="false"
                aria-labelledby="bp3-tab-title_template-details-parent_BASIC"
                class="bp3-tab-panel"
                id="bp3-tab-panel_template-details-parent_BASIC"
                role="tabpanel"
              >
                <div
                  class="Layout--vertical StyledProps--main"
                >
                  <div
                    class="StyledProps--font StyledProps--main"
                  >
                    <div
                      class="Layout--vertical Layout--layout-spacing-large StyledProps--main topContainer StyledProps--padding-top-xlarge StyledProps--padding-right-xxlarge StyledProps--padding-bottom-xlarge StyledProps--padding-left-xxlarge"
                    >
                      <div
                        class="StyledProps--font StyledProps--main"
                      >
                        <div
                          class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
                        >
                          <p
                            class="StyledProps--font StyledProps--main StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-black"
                          >
                            typeLabel
                          </p>
                          <div
                            class="StyledProps--font StyledProps--main"
                          >
                            <div
                              class="Layout--horizontal Layout--layout-spacing-small StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center StyledProps--flex-justifyContent-flex-start"
                            >
                              <span
                                class="bp3-icon StyledProps--main"
                                data-icon="harness-with-color"
                              >
                                <svg
                                  height="20"
                                  viewBox="0 0 25 25"
                                  width="20"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M22.974 8.186L16.708 1.88a7.023 7.023 0 00-2.59-1.572C11.98-.409 9.858.154 8.135 1.88L1.868 8.186a7.081 7.081 0 00-1.562 2.606c-.713 2.153-.153 4.289 1.562 6.023l6.266 6.306a7.113 7.113 0 002.589 1.58 5.663 5.663 0 001.808.308c1.503 0 2.946-.64 4.177-1.88l6.266-6.305a7.08 7.08 0 001.562-2.606c.713-2.162.153-4.298-1.562-6.032zm-10.07-4.477c.706.213 1.232.7 1.274.735l1.851 1.862-3.608 3.631-3.608-3.63 1.859-1.872c.5-.496 1.197-1.034 2.233-.726zm-9.219 8.305c.212-.71.696-1.24.73-1.282l1.85-1.863 3.609 3.632-3.608 3.63-1.86-1.87c-.492-.504-1.027-1.214-.721-2.247zm8.252 9.279c-.705-.214-1.231-.701-1.273-.735l-1.851-1.863 3.608-3.631 3.608 3.631-1.86 1.871c-.5.496-1.205 1.034-2.232.726zm9.22-8.305c-.212.709-.696 1.239-.73 1.281l-1.851 1.863-3.608-3.631 3.608-3.632 1.86 1.872c.492.504 1.027 1.204.72 2.247z"
                                    fill="currentColor"
                                    fill-rule="evenodd"
                                  />
                                </svg>
                              </span>
                              <p
                                class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey900"
                              >
                                Manual Approval
                              </p>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="StyledProps--font StyledProps--main"
                      >
                        <div
                          class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
                        >
                          <p
                            class="StyledProps--font StyledProps--main StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-black"
                          >
                            description
                          </p>
                          <p
                            class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey900"
                          >
                            Flink is a versatile framework, supporting many different deployment scenarios in a mix and match fashion.
                          </p>
                        </div>
                      </div>
                      <div
                        class="StyledProps--font StyledProps--main"
                      >
                        <div
                          class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
                        >
                          <p
                            class="StyledProps--font StyledProps--main StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-black"
                          >
                            tagsLabel
                          </p>
                          <div
                            class="StyledProps--font StyledProps--main"
                          >
                            <div
                              class="StyledProps--font StyledProps--main container"
                            >
                              <div
                                class="StyledProps--font StyledProps--main"
                              >
                                <div
                                  class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main tagsContainer"
                                >
                                  <span
                                    class="Tag--main"
                                  >
                                    <span
                                      class="bp3-tag tag"
                                      style="max-width: 80px;"
                                    >
                                      <span
                                        class="bp3-text-overflow-ellipsis bp3-fill"
                                      >
                                        QAR
                                      </span>
                                    </span>
                                  </span>
                                  <span
                                    class="Tag--main"
                                  >
                                    <span
                                      class="bp3-tag tag"
                                      style="max-width: 80px;"
                                    >
                                      <span
                                        class="bp3-text-overflow-ellipsis bp3-fill"
                                      >
                                        Internal 1
                                      </span>
                                    </span>
                                  </span>
                                  <span
                                    class="Tag--main"
                                  >
                                    <span
                                      class="bp3-tag tag"
                                      style="max-width: 80px;"
                                    >
                                      <span
                                        class="bp3-text-overflow-ellipsis bp3-fill"
                                      >
                                        Canary 1
                                      </span>
                                    </span>
                                  </span>
                                  <span
                                    class="Tag--main"
                                  >
                                    <span
                                      class="bp3-tag tag"
                                      style="max-width: 80px;"
                                    >
                                      <span
                                        class="bp3-text-overflow-ellipsis bp3-fill"
                                      >
                                        BLUE
                                      </span>
                                    </span>
                                  </span>
                                  <span
                                    class="Tag--main"
                                  >
                                    <span
                                      class="bp3-tag tag"
                                      style="max-width: 80px;"
                                    >
                                      <span
                                        class="bp3-text-overflow-ellipsis bp3-fill"
                                      >
                                        Tag A
                                      </span>
                                    </span>
                                  </span>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        class="StyledProps--font StyledProps--main"
                      >
                        <div
                          class="Layout--vertical Layout--layout-spacing-small StyledProps--main"
                        >
                          <p
                            class="StyledProps--font StyledProps--main StyledProps--font-weight-semi-bold StyledProps--color StyledProps--color-black"
                          >
                            templatesLibrary.createNewModal.versionLabel
                          </p>
                          <div
                            class="bp3-popover-wrapper DropDown--main"
                          >
                            <div
                              class="bp3-popover-target"
                            >
                              <div
                                class=""
                              >
                                <div
                                  class="Layout--horizontal StyledProps--main DropDown--dropdownButton DropDown--withBorder StyledProps--flex"
                                  data-testid="dropdown-button"
                                  style="width: 300px;"
                                >
                                  <div
                                    class="Layout--horizontal StyledProps--main DropDown--labelWrapper StyledProps--flex"
                                  >
                                    <p
                                      class="StyledProps--font Text--lineclamp StyledProps--main DropDown--label"
                                      data-testid="dropdown-value"
                                      style="--text-line-clamp: 1;"
                                    >
                                      templatesLibrary.stableVersion
                                    </p>
                                  </div>
                                  <div
                                    class="Layout--horizontal StyledProps--main DropDown--btnWrapper StyledProps--flex"
                                  >
                                    <span
                                      class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey400"
                                      data-icon="main-chevron-down"
                                    >
                                      <svg
                                        height="8"
                                        viewBox="0 0 14 8"
                                        width="8"
                                        xmlns="http://www.w3.org/2000/svg"
                                      >
                                        <path
                                          d="M13 0c-.3 0-.5.1-.7.3L7 5.6 1.7.3C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7l6 6c.2.2.4.3.7.3s.5-.1.7-.3l6-6c.2-.2.3-.4.3-.7 0-.6-.4-1-1-1z"
                                          fill="#4d5969"
                                        />
                                      </svg>
                                    </span>
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="StyledProps--font StyledProps--main innerTabsContainer"
                  >
                    <div
                      class="bp3-tabs Tabs--main Tabs--horizontal"
                    >
                      <div
                        class="bp3-tab-list"
                        role="tablist"
                      >
                        <div
                          class="bp3-tab-indicator-wrapper"
                          style="height: 0px; transform: translateX(0px) translateY(0px); width: 0px;"
                        >
                          <div
                            class="bp3-tab-indicator"
                          />
                        </div>
                        <div
                          aria-controls="bp3-tab-panel_template-details_INPUTS"
                          aria-disabled="false"
                          aria-expanded="false"
                          aria-selected="false"
                          class="bp3-tab"
                          data-tab-id="INPUTS"
                          id="bp3-tab-title_template-details_INPUTS"
                          role="tab"
                          tabindex="0"
                        >
                          templatesLibrary.templateInputs
                        </div>
                        <div
                          aria-controls="bp3-tab-panel_template-details_YAML"
                          aria-disabled="false"
                          aria-expanded="true"
                          aria-selected="true"
                          class="bp3-tab"
                          data-tab-id="YAML"
                          id="bp3-tab-title_template-details_YAML"
                          role="tab"
                          tabindex="0"
                        >
                          yaml
                        </div>
                        <div
                          aria-controls="bp3-tab-panel_template-details_REFERENCEDBY"
                          aria-disabled="true"
                          aria-expanded="false"
                          aria-selected="false"
                          class="bp3-tab"
                          data-tab-id="REFERENCEDBY"
                          id="bp3-tab-title_template-details_REFERENCEDBY"
                          role="tab"
                        >
                          templatesLibrary.referencedBy
                        </div>
                      </div>
                      <div
                        aria-hidden="false"
                        aria-labelledby="bp3-tab-title_template-details_YAML"
                        class="bp3-tab-panel"
                        id="bp3-tab-panel_template-details_YAML"
                        role="tabpanel"
                      >
                        <div
                          class="StyledProps--font StyledProps--main container"
                        >
                          <div
                            class="Layout--vertical Layout--layout-spacing-medium StyledProps--main"
                          >
                            <div
                              class="StyledProps--font StyledProps--main StyledProps--padding-top-large StyledProps--padding-right-xxlarge StyledProps--padding-bottom-large StyledProps--padding-left-xxlarge StyledProps--border StyledProps--border-bottom StyledProps--background-white"
                            >
                              <p
                                class="StyledProps--font StyledProps--main StyledProps--font-size-normal StyledProps--font-weight-bold StyledProps--color StyledProps--color-grey800"
                              >
                                template.yaml
                              </p>
                            </div>
                            <div
                              class="StyledProps--font StyledProps--main"
                              style="flex-grow: 1;"
                            >
                              <textarea>
                                template:
    name: manju-test-template-qq-12344
    identifier: manjutesttemplate
    versionLabel: v4
    type: Step
    projectIdentifier: Templateproject
    orgIdentifier: default
    description: Flink is a versatile framework, supporting many different deployment scenarios in a mix and match fashion.
    tags:
        QAR: ""
        Internal 1: ""
        Canary 1: ""
        BLUE: ""
        Tag A: ""
    spec:
        type: HarnessApproval
        timeout: &lt;+input&gt;
        spec:
            approvalMessage: &lt;+input&gt;
            includePipelineExecutionHistory: true
            approvers:
                userGroups: &lt;+input&gt;
                minimumCount: &lt;+input&gt;
                disallowPipelineExecutor: false
            approverInputs:
                - name: "1"
                  defaultValue: "1"
                - name: ttt
                  defaultValue: ttt
        failureStrategies:
            - onFailure:
                  errors:
                      - Timeout
                  action:
                      type: MarkAsSuccess

                              </textarea>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    data-testid="location"
  >
    /
  </div>
</div>
`;
